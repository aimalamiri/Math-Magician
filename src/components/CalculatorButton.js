import React from 'react';
import PropTypes from 'prop-types';

export default class CalculatorButton extends React.PureComponent {
  render() {
    const { text } = this.props;
    return (
      <button className="btn" type="button">
        {text}
      </button>
    );
  }
}

CalculatorButton.propTypes = {
  text: PropTypes.string.isRequired,
};
