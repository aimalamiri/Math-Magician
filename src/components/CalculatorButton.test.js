import renderer from 'react-test-renderer';
import CalculatorButton from './CalculatorButton';

it('CalculatorButton renders correctly', () => {
  const handleClick = () => true;
  const tree = renderer.create(<CalculatorButton text="Submit" handleClick={handleClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});
