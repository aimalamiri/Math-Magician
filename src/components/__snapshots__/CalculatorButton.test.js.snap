// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton renders correctly 1`] = `
<button
  className="btn "
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;
